modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache

```diff
@@ -382,6 +382,14 @@ export class {{classname}} extends BaseAPI implements {{classname}}Interface {
 {{^withInterfaces}}
 export class {{classname}} extends BaseAPI {
 {{/withInterfaces}}
+    constructor(configuration: Configuration, basePath?: string, axios?: AxiosInstance) {
+        super(configuration, basePath, axios);
+        // bind function to keep `this` context
+        {{#operation}}
+        this.{{nickname}} = this.{{nickname}}.bind(this);
```

Comment:
Is there a reason to prefer doing it in the class rather than in the application code? in your example it could be something like
```
logWrapper(api.getPetById.bind(api), 1).then((result) => {
    console.log("Result: ", result);
});
```

E.g you wouldn't expect that to work with standard classes you consume in your app
```
const date = new Date()
logWrapper(date.setYear, 2021) // oops, run-time error
```

So what it so special about the API classes here that they need special treatment?

Reply:
Fair point. 
I honestly had forgotten about the strange behavior of `this` in javascript, and was surprised by the errors.
Having it in the class keeps the behavior the same between all the ways to call the api, saves a bit of typing, and helps other less-experienced developpers not run into issues 😄

Reply:
+1 to @amakhrov s comment, the context binding is very standard in javascript, closing this PR